Remove gas cost and estimate gas limit

luckysori opened the pull request:
Resolves #491.

Commit:
Remove gas cost/price

Commit:
Estimate gas limit for deployment and send

- Add `deployment_gas_limit` and `transaction_gas_limit` to trait
  `HTLC` and implement it for `EtherHtlc` and `Erc20Htlc`.
- Use `deployment_gas_limit` and `transaction_gas_limit` instead of
  hard-coded values when creating `ContractDeploy` and
  `SendTransaction`.

LLFourn reviewed (changes requested):
Nice work. Just some re-org to do and to calculate ERC20 call cost.

thomaseizinger reviewed (changes requested):
I agree with Lloyd, just make it methods on the HTLC itself :)

Otherwise, good work! Some missing newlines though :upside_down_face:

Commit:
Refactor and introduce fund_gas_limit

- Remove `deployment_gas_limit` and `transaction_gas_limit` from
  `Htlc` trait.
- Reintroduce `deployment_gas_limit` and `transaction_gas_limit`
  in `EtherHtlc` impl block. `deployment_gas_limit` now is a
  method that takes `&self`.
- Reintroduce `deployment_gas_limit` in `Erc20Htlc` impl block.
- Add `redemption_gas_limit` and `fund_gas_limit`, and reintroduce
  `transaction_gas_limit` as private method in `EtherHtlc` impl block.
  `redemption_gas_limit` just calls `transaction_gas_limit`, whereas
  `fund_gas_limit` is 4 times `transaction_gas_limit`.

Commit:
Simplify estimates
